### ISSUE TITLE
[RFE] Add Support for "resourceNames" in Rancher's RBAC

### ISSUE THREAD

**Issue by mrdima** [rancher/rancher#38368]
This is literally a continuation of issue #31329 from @asemen2608 which was closed due to stale bot, but still relevant in Rancher 2.6.6. Part of the old issue's text pasted.

Feature request:
The resourceNames field should be exposed and usable when creating custom roles in Rancher.

Our specific usecase of using resourceNames in RBAC is giving 'get' rights to specific clusterroles and also to specific crd's.

Rancher role templates are just Kubernetes roles. As such, whilst the Rancher UI does not expose the resourceNames field of rules, this field exists on the API object (schema can be seen at https://<rancher_url>/v3/schemas/policyRule).
Therefore, a user can add or edit resourceNames for a role, within the API Explorer, by clicking 'View in API' for a role from the Rancher UI.

Other usecase in old issue
Limit at Rancher's level RBAC to prevent people from deleting specific namespaces with a given name.
In Kubernetes, that would be translated to:

rules:

apiGroups: [""]
resources: ["namespaces"]
resourceNames: ["cattle-system"]
verbs: ["list", "get", "watch"]
However, the particular field "resourceNames" isn't visible for end-users in Rancher.

**Comment by SanderHuijbers** [comment]
I totally agree on this. It is now not possible to give cluster members more permissions on their own resources like crd's, without opening access to all other crd's. This just one example we experience too.
